src/code/z_actor.c

```diff
@@ -908,41 +908,38 @@ s32 func_800B6434(PlayState* play, TitleCardContext* titleCtx) {
     return true;
 }
 
-// ActorContext_1F4 Init
-void func_800B6468(PlayState* play) {
-    play->actorCtx.unk_1F4.timer = 0;
+void Actor_InitPlayerImpact(PlayState* play) {
+    play->actorCtx.playerImpact.timer = 0;
 }
 
-// ActorContext_1F4 Update
-void func_800B6474(PlayState* play) {
-    DECR(play->actorCtx.unk_1F4.timer);
+void Actor_UpdatePlayerImpact(PlayState* play) {
+    DECR(play->actorCtx.playerImpact.timer);
 }
 
-// ActorContext_1F4 setter something
-s32 func_800B648C(PlayState* play, s32 arg1, s32 timer, f32 arg3, Vec3f* arg4) {
-    if ((play->actorCtx.unk_1F4.timer != 0) && (arg3 < play->actorCtx.unk_1F4.unk_04)) {
+s32 Actor_SetPlayerImpact(PlayState* play, PlayerImpactType type, s32 timer, f32 dist, Vec3f* pos) {
+    if ((play->actorCtx.playerImpact.timer != 0) && (dist < play->actorCtx.playerImpact.dist)) {
         return false;
     }
 
-    play->actorCtx.unk_1F4.unk_00 = arg1;
-    play->actorCtx.unk_1F4.timer = timer;
-    play->actorCtx.unk_1F4.unk_04 = arg3;
-    Math_Vec3f_Copy(&play->actorCtx.unk_1F4.unk_08, arg4);
+    play->actorCtx.playerImpact.type = type;
+    play->actorCtx.playerImpact.timer = timer;
+    play->actorCtx.playerImpact.dist = dist;
+    Math_Vec3f_Copy(&play->actorCtx.playerImpact.pos, pos);
 
     return true;
 }
 
-// ActorContext_1F4 getter something
-f32 func_800B64FC(PlayState* play, f32 arg1, Vec3f* arg2, u32* arg3) {
-    f32 temp_f8;
+f32 Actor_GetPlayerImpact(PlayState* play, f32 range, Vec3f* pos, PlayerImpactType* type) {
+    f32 dist;
 
-    if ((play->actorCtx.unk_1F4.timer == 0) || (arg1 == 0.0f)) {
+    if ((play->actorCtx.playerImpact.timer == 0) || (range == 0.0f)) {
         return -1.0f;
     }
 
-    temp_f8 = Math_Vec3f_DistXYZ(&play->actorCtx.unk_1F4.unk_08, arg2) / arg1;
-    *arg3 = play->actorCtx.unk_1F4.unk_00;
-    return play->actorCtx.unk_1F4.unk_04 - temp_f8;
+    dist = Math_Vec3f_DistXYZ(&play->actorCtx.playerImpact.pos, pos) / range;
+    *type = play->actorCtx.playerImpact.type;
+
+    return play->actorCtx.playerImpact.dist - dist;
```

Comment:
I think the interpretation of `range` is incorrect, then: it's more like a scale factor, and `playerImpact.dist` is the maximum range? It seems like the way this is supposed to be used is you check if the return is `>= 0.0f` (if so, would be useful to indicate that).

Reply:
All uses of this function do check the return of `>= 0.0f` including from Boss07, although in theory you could add an offset for the range by checking `>= offset` or something.

It is indeed scaling the distance, but can that not be interpreted as changing the range? If you double the `range` parameter, then your effective distance is half meaning you can feel the effect of the player impact twice as far while still being bound by `playerImpact.dist`.

Something like `playerImpact.distMax` might work, but then it might give the wrong impression that it's a hard limit, when it's more `playerImpact.distMaxRelToRange` or `playerImpact.distMaxRelToScale`